File: Lift.Buddy.UI/ClientApp/src/app/Model/PersonalRecord.ts
import { ExerciseType } from "./Enums/ExerciseType";
import { UnitOfMeasure } from "./Enums/UnitOfMeasures";

export class PersonalRecord {
    public exerciseName: string = '';
    public series: number = 0;
    public reps: number = 0;
    public weight: number = 0;
    public unitOfMeasure: UnitOfMeasure = UnitOfMeasure.KG;
    public exerciseType: ExerciseType = ExerciseType.Weight;
    public id: string | null = null;
    public userId: string | null = null;
}

File: Lift.Buddy.Core/Models/PersonalRecord.cs
using Lift.Buddy.Core.Models.Enums;

namespace Lift.Buddy.Core.Models
{
    //TODO: mettere ExerciseName, ExcerciseType in Excercise ed usare quello?
    public class PersonalRecord
    {
        public string ExerciseName { get; set; } = string.Empty;

        public int Series { get; set; }
        public int Reps { get; set; }

        public int Weight { get; set; }

        //TODO: enum al posto di string? magari un oggetto "Weight" che ha sia peso che UOM
        public string UnitOfMeasure { get; set; } = "KG";

        public ExercizeType ExerciseType { get; set; }
    }
}
